### ISSUE TITLE
Health calculation rework

### ISSUE THREAD

**Issue by GizzleBizzle** [The-Alpha-Project/alpha-core#1147]
**Describe the bug**
Currently all classes get +10 HP per stamina. That wasn't always the case as I'm about to prove.

**Steps to reproduce**
1. Put on any stamina gear/buff on the character

**Expected behavior**
One point of stamina should result in:
20 additional HP for warriors
18 for paladins
16 for rogues, warlocks and shamans
14 for mages and priests

Because 0.6 changed this method,of calculating hp from stamina - Druids and Hunters couldn't be researched, as they became playable in 0.6 and 0.9 respectively. It is up to @GrenderG to decide how many hp they should get. Personally I believe they should be given 16 points per stam, since 14hp level seems to be reserved only for pure caster classes.

**Proof**
[Some items have both stamina and flat HP stats](https://db.thealphaproject.eu/index.php?action=show_item&id=4998&filter=blood+ring&sort_order=name&pos=1&max=1) 
[Forum talk about changing the formula yet again in 0.10 ](https://crawler.thealphaproject.eu/mnt/crawler/media/Forum/WoW%20Beta%20Official%20Forum%20Dump%202004/class/thread.aspx%3Ffn=wow-warrior&t=7031&p=1.html) - Orlaba's post
[Forum conversation mentioning health calculation standarization around 0.11](https://crawler.thealphaproject.eu/mnt/crawler/media/Forum/WoW%20Beta%20Official%20Forum%20Dump%202004/class/thread.aspx%3Ffn=wow-druid&t=7491&p=1&.html) - GaboTheGreat's post
[Chat conversation about hp values from stamina](https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Eastern%20Kingdoms/Dun%20Morogh/741.jpg)

And finally...
[My GoogleDoc, where I've done the calculations ](https://docs.google.com/spreadsheets/d/1tOgZ4qSakVoV8CrFGI1pNLnjUh-D5BGqMvx3z9y9DuA/edit?usp=sharing) - It will serve as a tool for the next project - tracking attribute changes throughout alpha lvling process. Don't worry if it looks unpolished. It will change in time.

**Comment by GrenderG** [comment]
![image](https://github.com/The-Alpha-Project/alpha-core/assets/8654362/37f0e3b7-b933-49a0-af0e-e0c617d301ed)
Interesting find. Where does the info regarding 14 for mages/priests come from?

**Comment by GrenderG** [comment]
![image](https://github.com/The-Alpha-Project/alpha-core/assets/8654362/037d7236-6c4f-4c07-8f05-6481b790a67b)
Some more chatting mentioning different hp/stamina per class.

**Comment by GizzleBizzle** [comment]
The druid's screenshot is from a revamped health calculation introduced in 0.6. The one I researched was from 0.5.5 and quite possibly 0.5.4 aswell, since what few screenshots we have from that patch seem to match my calculations.

As for the hp per stamina:

The base hp pool is 40 hp and you get it for the first 20 points of stamina on the character. Only after getting subsequent stamina points the calculation starts working.

I found out about this from lvl 1 warrior character sheet screenshots.
Dwarf warrior starts with 23 stam and 100 hp (40 hp base + (3×20hp from stamina)).
Human warriors start with 22 stam and 80 hp (40 + (2×20)).
Night elves start with 21 stam and 60 hp... you get the picture.

This was later proven for other classes. Proof in the GoogleDoc. The base hp for all classes is 40hp.

So the rest of the work was to find out as many screenshots where attributes were present and do some maths. 

For instance - [A human mage ](https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Eastern%20Kingdoms/Elwynn%20Forest/0320wow05.jpg) starts with 68 hp. Substract the base 40 hp and divide by 2 points of stam... voila.

Priest[Priest was harder](https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Eastern%20Kingdoms/Elwynn%20Forest/HumanAlphaCharScreen.jpg). I only found a lvl 2 one. Did more research and discovered  that the flat mana and hp values gained from lvlups of a class are the same for all races of this class (both gnomes, humans and every other race will get +8 hp and +12 mana when reaching lvl 2 as a mage).

I found [This](https://archive.thealphaproject.eu/media/Alpha-Project-Archive/Images/Azeroth/Kalimdor/Teldrassil/wowb_031804_006.jpg) and simply subtracted hp and mana. The result was 68 hp with 22 stam. The calculation was the same as priest's.
